cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1244,17 +1247,32 @@ table_with_metadata reader::impl::read(size_type skip_rows,
   return {std::make_unique<table>(std::move(out_columns)), std::move(out_metadata)};
 }
 
+table_with_metadata reader::impl::read(parquet_reader_options const &options, cudaStream_t stream)
+{
+  auto skip_rows         = options.skip_rows();
+  auto num_rows          = options.num_rows();
+  auto const &row_groups = options.row_groups();
+
+  if (row_groups.size() > 0) {
+    return read(0, -1, row_groups, stream);
+  } else if ((skip_rows != -1) || (num_rows != -1)) {
+    return read(skip_rows, (num_rows != 0) ? num_rows : -1, {}, stream);
+  } else {
+    return read(0, -1, {}, stream);
+  }
```

Comment:
But in a case where `row_groups` is available and even `skip_rows` and `num_rows` has some value, then it wouldn't be same as it is now.

Reply:
Current implementation means - ignore `skip_rows` and `num_rows` if `row_groups` is set. I'm not sure if this is intentional, there's documentation to suggest so.
Whatever the target behavior is, I'd prefer for it be enforced by making sure the options object is always valid (now that we can).
If we want to enforce the current behavior, `parquet_reader_options` setters should ignore `skip_rows` and `num_rows` calls (or throw?) once `row_groups` is set, and reset `skip_rows` and `num_rows` in the `row_groups` setter.
If we want to treat them as equal, they should reset each other in setters (or throw if other is already set).
IIRC, CSV reader throws if you use multiple row selection options at the same time.

Alternatively, maybe we could design the builder API so it's more obvious that different row selection options would cancel each other out. Something like
```
opt& select_rows(size_type skip, size_type num);
opt& select_rows(vector<int> row groups); // call sites might not be clear, since group is not in the API name
```

Reply:
I would go for throw, as it will be easier for developer to track it and would be easier to understand.

Reply:
yay, we found something for setters to do :)
I'm fine with throwing when setting to a non-default value.